### DISCUSSION TITLE
"-page" behaves differently in v6 vs v7

### DISCUSSION THREAD

**Discussion by manuel-koch** [ImageMagick/ImageMagick#7607] [Help] [answered]
### ImageMagick version

7.1.1-38 & 6.9.10-23

### Operating system, version and so on

MacOS & Linux

### Description

I'm trying to stitch three input images in one output image.
Running almost the same commands ( only executable-name and font differs ) on two systems
- on Linux: ImageMagick 6.9.10-23 Q16 x86_64 20190101
- on MacOS: ImageMagick 7.1.1-38 Q16-HDRI aarch64 22398

gives me completely different results and I don't know why or how to get same results.

Any help/hint appreciated !

**Linux: ImageMagick 6.9.10-23 Q16 x86_64 20190101**
```
convert -colorspace rgb \
    -page +859+32 page-left.png -page +827+0 -background blue -fill white -font "DejaVu-Sans" -size x32 label:left \
    -page +859+32 page-right.png -page +891+0 -background blue -fill white -font "DejaVu-Sans" -size x32 label:right \
    -page +0+32 page-middle.png -page +32+0 -background blue -fill white -font "DejaVu-Sans" -size x32 label:middle \
    -background white -layers merge +repage page-collection-v6.png
```
results in
![page-collection-v6](https://github.com/user-attachments/assets/f8f9494c-d19b-4f40-a45f-813326736aa4)

**MacOS: ImageMagick 7.1.1-38 Q16-HDRI aarch64 22398**
```
magick -colorspace rgb \
    -page +859+32 page-left.png -page +827+0 -background blue -fill white -font Courier -size x32 label:left \
    -page +859+32 page-right.png -page +891+0 -background blue -fill white -font Courier -size x32 label:right \
    -page +0+32 page-middle.png -page +32+0 -background blue -fill white -font Courier -size x32 label:middle \
    -background white -layers merge +repage page-collection-v7.png
```
results in
![page-collection-v7](https://github.com/user-attachments/assets/364c848a-375e-46af-a8d1-a9f5d2b9ad7f)

Using these three input images:
![page-left](https://github.com/user-attachments/assets/f9bb23c5-2262-4459-9229-56a9dbb9de2e)
![page-middle](https://github.com/user-attachments/assets/cfb4c07c-b946-4b90-9349-2f5d091afce0)
![page-right](https://github.com/user-attachments/assets/f473bcdf-9428-4e18-9ac9-30cfef30c575)

**Comment by fmw42** [comment]
This was explained in the porting notes at https://imagemagick.org/script/porting.php#cli.  See Behavioral Changes. It was intended due to the way IM 7 is different from IM 6.

**Comment by manuel-koch** [comment, marked as answer]
After refactoring command line to stitch images the following works and results in equivalent output.
The trick in my use case is to enforce `-gravity` and put `-page` fragments in parentheses.

**MacOS: ImageMagick 7.1.1-38 Q16-HDRI aarch64 22398**
```
magick -colorspace rgb \
    -gravity NorthWest \
    \( -page +0+32 page-left.png \) \
    \( -page +32+0 -background blue -fill white -font Courier -size x32 label:left \) \
    \( -page +859+32 page-middle.png \) \
    \( -page +891+0 -background blue -fill white -font Courier -size x32 label:middle \) \
    \( -page +1718+32 page-right.png \) \
    \( -page +1750+0 -background blue -fill white -font Courier -size x32 label:right \) \
    -background white -layers merge +repage page-collection.png
```
